### PR TITLE
fix(engine): double quotes in variables should be correctly handled

### LINKED ISSUE #16043: Regression in 8.4: Failed to build variable mapping expression
**Describe the bug**

I can't deploy a BPMN process to Zeebe 8.4.0. In previous versions (8.3.5), I deployed the process successfully. The deployment reports the following error:

```
Command 'CREATE' rejected with code 'INVALID_ARGUMENT': Expected to deploy new resources, but encountered the following errors:
'New BPMN Diagram.bpmn': Failed to build variable mapping expression: failed to parse expression '{apiKey:key,from:{name:"Me",email:"me@camunda.com"},to:{name:""You"",email:"you@camunda.com"},unMappedFieldNotUseInModel:{mailType:"mail"},content:{subject:"Hello",type:"text/plain",value:"This is a test."}}': Expected (start-of-input | ifOp | forOp | quantifiedOp | disjunction):1:1, found "{apiKey:ke"[ deploy-error ] 23/1/2024 8:37:14
``` 

Screenshot from 2024-01-23 08-40-09

The process contains a SendGrid connector task with the configuration:
- Receiver Name = `"You"` (static value but with double-quotes)

**To Reproduce**

1. Deploy the BPMN process [regression-variable-mapping.bpmn](https://github.com/camunda/zeebe/files/14019955/regression-variable-mapping.bpmn.txt)
2. Verify that the deployment fails

**Expected behavior**

I can deploy the BPMN process.

If the static value is wrapped in double quotes then the double quotes are removed on parsing. When the variable mapping is applied, the process variable should be a string value `You` without double quotes (i.e. as JSON `"You"` and not `"\"You\""`).

**Log/Stacktrace**

This issue was reported as a [Support case](https://jira.camunda.com/browse/SUPPORT-20145). 

**Environment:**
- OS: 
- Zeebe Version: `8.4.0`
- Configuration:

### PR DESCRIPTION
**Pull request by nicpuppa**
## Description

This PR fix a wrong behaviour introduced in `8.4`. If the static value is wrapped in double quotes then the double quotes should be escaped on parsing. 

## Related issues

closes #16043 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/transformer/VariableMappingTransformer.java [hunk 1]
```diff
@@ -154,7 +154,8 @@ public String onEntry(final String targetKey, final Expression sourceExpression)
         final String expression;
 
         if (sourceExpression instanceof StaticExpression) {
-          expression = String.format("\"%s\"", sourceExpression.getExpression());
+          expression =
+              String.format("\"%s\"", sourceExpression.getExpression().replaceAll("\"", "\\\\\""));
```
Comment (saig0) [inline engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/transformer/VariableMappingTransformer.java:158]: :wrench: Please add a comment that explains what is happening here and why. :sweat_smile:

File: engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/ActivityTest.java [hunk 2]
```diff
@@ -78,7 +78,7 @@ public final class ActivityTest {
                       .zeebeInput("https://github.com/{{orgId}}/{{repoId}}", "urlStatic")
                       .zeebeInputExpression(
                           "\"https://github.com/{{orgId}}/{{repoId}}\"", "urlExpression")
-                      .zeebeInput("My Name is &#34;Zeebe&#34;, nice to meet you", "quotesStatic")
```
Comment (saig0) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/ActivityTest.java:81]: :wrench: Let's keep both test cases: `\"` and `&#34;`. The Modeler uses `&#34;` for double quotes.

File: engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/ActivityTest.java [hunk 3]
```diff
@@ -228,6 +228,41 @@ public void shouldIgnoreTaskHeadersIfNull() {
     createProcessAndAssertIgnoredHeaders(null);
   }
 
+  // regression test for https://github.com/camunda/zeebe/issues/16043
+  @Test
+  public void shouldApplyTheCorrectEscapesToVariables() {
```
Comment (saig0) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/ActivityTest.java:233]: :wrench: 

```suggestion
  public void shouldEscapeDoubleQuotesInStaticExpression() {
```

File: engine/src/test/java/io/camunda/zeebe/engine/processing/variable/mapping/VariableMappingTransformerTest.java [hunk 4]
```diff
@@ -106,6 +106,7 @@ public void shouldNotEscapeCharactersInStaticExpression() {
                 mapping("Hello\tWorld", "tab"),
                 mapping("Hello\nWorld", "newline"),
                 mapping("Hello\rWorld", "carriageReturn"),
+                mapping("\"My Name is \"\"Zeebe\"\", nice to meet you\"", "doubleQoutes"),
```
Comment (saig0) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/variable/mapping/VariableMappingTransformerTest.java:109]: :thinking: Why do we have two double quotes `\"\"Zeebe\"\"`?

**Review by saig0 — APPROVED** [review]
@nicpuppa nice. :tada:  I did a manual test and can confirm that it is behaving again like in 8.3. :rocket: 

I have a few minor comments. Please have a look before merging. :+1:

File: engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/transformer/VariableMappingTransformer.java (hunk 1 above)
Comment (nicpuppa) [reply engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/transformer/VariableMappingTransformer.java:158]: We don't know what is happening here 🤣

File: engine/src/test/java/io/camunda/zeebe/engine/processing/variable/mapping/VariableMappingTransformerTest.java (hunk 4 above)
Comment (nicpuppa) [reply engine/src/test/java/io/camunda/zeebe/engine/processing/variable/mapping/VariableMappingTransformerTest.java:109]: Double quotes made me crazy 🤣

**Comment by nicpuppa** [comment]
Hey @saig0 I applied your suggestion, I will merge the PR after the CI 🚀 👍

**Pull request merged**